lib/matplotlib/__init__.py

```diff
@@ -954,7 +970,10 @@ def _rc_params_in_file(fname, fail_on_error=False):
 http://github.com/matplotlib/matplotlib/blob/master/matplotlibrc.template
 or from the matplotlib source distribution""" % (key, cnt, fname),
                   file=sys.stderr)
-
+    # setting the backend will force this to false, but we should respect
+    # the users input here so re-set it.
+    if try_backend_fallback_again is not None:
+        config['backend_fallback'] = try_backend_fallback_again
```

Comment:
this logic is so ridiculously complicated :(

Reply:
Yeah I don’t quite understand all the use cases we are trying to accommodate here.  But I’d sure be in favour of simpler logic and better error messages.